### DISCUSSION TITLE
Comparison between utility-first CSS framework

### DISCUSSION THREAD

**Discussion by kafai-lam** [master-co/css#3] [Question]
What is the main difference if we compare master styles to other utility CSS framework [tailwindcss](https://github.com/tailwindlabs/tailwindcss), [windicss](https://github.com/windicss/windicss), or [unocss](https://github.com/unocss/unocss), in the perspective of fundamental idea, mechanism, performance?

**Comment by 1aron** [comment]
I will write a detailed comparison in the future depending on the situation.

1. The syntax of Master is completely different from the three frameworks you mentioned, all of which are original by us, and are almost close to native CSS
2. TailwindCSS and WindiCSS are currently not able to JIT in production browsers.
3. `@unocss` is very different from `@master/css` in terms of implementation and essence. I also read its source code after a developer in Taiwan mentioned it. They are a basic conversion engine, designed for the current well-known Bootstrap, TailwindCSS, WindiCSS. When using selectors, media must be defined one by one, but our intelligent algorithm allows you to unlock all features without definition, we do smart rule transformations for native CSS, so I call it Virtual CSS.

By the way, I personally appreciate the open source projects from the authors of unocss, they are all quite interesting and thought-provoking ! 😍
